src/main/java/com/autotune/analyzer/workerimpl/BulkJobManager.java

```diff
@@ -98,176 +100,271 @@ public static List<String> appendExperiments(List<String> allExperiments, String
+            dateRange.put("end_time", interval_end_time_epoc);
+            dateRange.put("steps", steps);
+        }
+        return dateRange;
+    }
+
+
+
+    /**
+     * @param dc         DataSourceContainer object to get the container details
+     * @param dsc        DataSourceCluster object to get the cluster details
+     * @param dsw        DataSourceWorkload object to get the workload details
+     * @param namespace  DataSourceNamespace object to get the namespace details
+     * @param datasource Datasource name to be set
+     * @return Json string to be sent to the createExperimentAPI for experiment creation
+     * @throws JsonProcessingException
+     */
+    private String prepareCreateExperimentJSONInput(DataSourceContainer dc, DataSourceCluster dsc, DataSourceWorkload dsw,
+                                                    DataSourceNamespace namespace, String experiment_name, String datasource) throws IOException {
+
+        CreateExperimentConfigBean createExperimentConfigBean = CREATE_EXPERIMENT_CONFIG_BEAN;
+        // Experiment name
+        createExperimentConfigBean.setExperiment_name(experiment_name);
+        // Datasource
+        createExperimentConfigBean.setDatasource(datasource);
+        // Cluster name
+        createExperimentConfigBean.setCluster_name(dsc.getDataSourceClusterName());
+        // Kubernetes objects
+        List<KubernetesAPIObject> kubernetesAPIObjectList = new ArrayList<>();
+        KubernetesAPIObject kubernetesAPIObject = new KubernetesAPIObject();
+        kubernetesAPIObject.setType(dsw.getDataSourceWorkloadType());
+        kubernetesAPIObject.setName(dsw.getDataSourceWorkloadName());
+        kubernetesAPIObject.setNamespace(namespace.getDataSourceNamespaceName());
+        // Containers
+        ContainerAPIObject containerAPIObject = new ContainerAPIObject(dc.getDataSourceContainerName(),
+                dc.getDataSourceContainerImageName(), null, null);
+        // Add container to the Kubernetes object
+        kubernetesAPIObject.setContainerAPIObjects(Arrays.asList(containerAPIObject));
+        kubernetesAPIObjectList.add(kubernetesAPIObject);
+        // Add the Kubernetes objects to the createExperimentConfigBean
+        createExperimentConfigBean.setKubernetes_objects(kubernetesAPIObjectList);
+
+        // list to hold CreateExperimentConfigBean objects
+        List<CreateExperimentConfigBean> createExperimentConfigBeanList = new ArrayList<>();
+
+        // Add CreateExperimentConfigBean object to the list
+        createExperimentConfigBeanList.add(createExperimentConfigBean);
+
+        // Convert to JSON
+        ObjectMapper mapper = new ObjectMapper();
+        String json = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(createExperimentConfigBeanList);
+        LOGGER.debug("CreateExp JSON: {}", json);
+        return json;
+    }
+
+    public static int sendCreateExperimentRequest(String jsonInput) throws IOException {
+        URL url;
```

Comment:
Can we make use of GenericRestAPIClient Class Please see if this can be used for geenratingrecommendations URL too